docs: Updated and expand BN hardware specs (#2382)
hiero-ledger/hiero-block-node: docs/block-node/operations/block-node-hardware-specifications.md

|   Component    |                   Minimum Specification                   |
|----------------|-----------------------------------------------------------|
| CPU            | 24 cores / 48 threads, 2024 or newer (PCIe 4+), ≥ 2.0 GHz |
| RAM            | 256 GB                                                    |
| Fast NVMe Disk | 8 TB NVMe SSD                                             |
| Network        | 2 × 10 Gbps NICs                                          |
| OS             | Linux (Ubuntu 22.04 LTS or Debian 11 LTS recommended)     |

**Recommendations:**

- NICs: 20 Gbps or higher are recommended for better throughput and future-proofing,
  even though 10 Gbps is the stated minimum.
- Bulk storage: 500 TB is recommended for LFH to accommodate long-term block history
  and state growth. 300 Tb would be the next preferred and 100 TB the minimum for Tier 1.
- Servers may be sourced from bare metal providers or cloud providers offering dedicated
  instances. LFH configurations require significant storage capacity and are typically
  sourced from bare metal providers.
- OS disk requirements are minimal; the OS disk sees little activity after start-up and

>   does not require explicit sizing beyond standard OS installation needs.

Reviewer: Some policies truly minimize O/S disk, so we need to make sure there is room for OCI image files.
```suggestion
- OS disk requirements are minimal; the OS disk sees little activity after start-up and
  does not require explicit sizing beyond standard OS installation needs and at least 10GB for OCI image storage.
```

---

## Storage Benchmark Targets

The Block Node is I/O-intensive. The following benchmarks define the sustained throughput,
IOPS, and latency targets that storage must meet to avoid becoming a bottleneck.

### Disk Performance Targets

| Disk Type  | Sustained Write | Sustained Read | Required Write IOPS | Required Read IOPS | P99 Write Latency | P99 Read Latency |
|------------|-----------------|----------------|---------------------|--------------------|-------------------|------------------|

> | Fast NVMe  | 4.8 Gbps        | 19.8 Gbps      | 250k                | 800k               | < 300 µs          | < 200 µs         |

Author: Units should be right
For instance taking todays 10k TPS block of around 21MB. Apply a 40% overhead and round up for nice numbers give you 30 MB.
That block today is across 2 sec, so double the amount to cover future scaling option of that same amount at 1 sec (I know we're aiming for more but this is a balance).
Then we're using a clean factor of 10 to go from MBps to Mbps to account for ethernet transfers and the extra bits.
Then not the consideration of unlikely but worst case scenario of around 8 CNs pushing to a single block node - assumes smallest distribution of 39 CNs and 5 BNs which shouldn't happen.

So 21 MB * 1.4 * 2 * 10 bits/byte * 8 ~= 4704 Gbps => rounded up is 4.8 Gbps

I didn't want to call out to much details here but I can if you think it'd be helpful

### Ingress Bandwidth (Consensus Node → Block Node)

Steady-state ingress carries one uncompressed block stream per second.
Worst-case reflects 8 Consensus Nodes simultaneously streaming to a single BN.

|    TPS | Steady-state ingress | Worst-case ingress (8× catch-up) |
|-------:|---------------------:|---------------------------------:|
|  2,000 |             ~60 Mbps |                        ~480 Mbps |
| 20,000 |            ~600 Mbps |                      ~4,800 Mbps |

> At 20K TPS worst-case, ingress alone approaches the 10 Gbps NIC minimum.
> A 20 Gbps NIC (or bonded pair) is **strongly recommended** for any deployment
> expected to operate at ≥ 10K TPS or to backfill aggressively.

### Egress Bandwidth (Block Node → Subscribers)

Each downstream subscriber (Mirror Node, Block Node, DApp) receives its own uncompressed stream.

|    TPS | Per subscriber / day | Per subscriber / month | 33 subscribers / day | 33 subscribers / month |
|-------:|---------------------:|-----------------------:|---------------------:|-----------------------:|
|  2,000 |               6.5 TB |                 196 TB |               215 TB |                 6.5 PB |

> | 20,000 |                65 TB |                 1.9 PB |             2,145 TB |                64.4 PB |

Reviewer: We can set HTTP (GZip) compression on the PBJ plugin, and probably should.
Also, I think you added a large multiplier for single streams somewhere that carried forward.  If we're delivering 6+ petabytes monthly at 2K TPS then block nodes are non-viable.

```suggestion
Each downstream subscriber (Mirror Node, Block Node, DApp) receives its own stream.

|    TPS | Per subscriber / day | Per subscriber / month | 33 subscribers / day | 33 subscribers / month |
|-------:|---------------------:|-----------------------:|---------------------:|-----------------------:|
|  2,000 |               83 GB |                 2.5 TB |               2.7 TB |                 82 TB |
| 20,000 |               747 GB |                22.4 TB |             24.6 TB |                738 TB |
```

**Worst-case bandwidth peaks:**

|    TPS | Worst-case egress (33 subscribers) |
|-------:|-----------------------------------:|
|  2,000 |                        ~2,000 Mbps |
| 20,000 |                       ~20,000 Mbps |

> At 20K TPS with 33 subscribers (1/3 of max BNs (13) backfilling, 10 MNs and 10 DApps subscribed),
> egress at ~20 Gbps saturates the 10 Gbps minimum NIC.
> Block Nodes serving many live subscribers at high TPS **require** at least a 20 Gbps NIC

> > and may need 40 Gbps or bonded links for headroom.

Reviewer: ```suggestion
> Block Nodes serving many live subscribers at high TPS **require** at least a 25 Gbps NIC
> and may need 100 Gbps or multiple bonded 25 Gbps links for headroom.
```
